community/neo4j/src/test/java/schema/IndexPopulationIT.java

```diff
@@ -132,6 +137,42 @@ public void indexCreationDoNotBlockWritesOnOtherLabel() throws ExecutionExceptio
         }
     }
 
+    @Test
+    public void shutdownDatabaseDuringIndexPopulations()
+    {
+        AssertableLogProvider assertableLogProvider = new AssertableLogProvider( true );
+        File storeDir = directory.directory( "shutdownDbTest" );
+        Label testLabel = Label.label( "testLabel" );
+        String propertyName = "testProperty";
+        GraphDatabaseService shutDownDb = new TestGraphDatabaseFactory().setInternalLogProvider( assertableLogProvider )
+                                                                      .newEmbeddedDatabase( storeDir );
+        prePopulateDatabase( shutDownDb, testLabel, propertyName );
+
+        try ( Transaction transaction = shutDownDb.beginTx() )
+        {
+            shutDownDb.schema().indexFor( testLabel ).on( propertyName ).create();
+            transaction.success();
+        }
+        shutDownDb.shutdown();
+        assertableLogProvider.assertNone( AssertableLogProvider.inLog( IndexPopulationJob.class ).anyError() );
+    }
+
+    private void prePopulateDatabase( GraphDatabaseService database, Label testLabel, String propertyName )
+    {
+        for ( int j = 0; j < 100; j++ )
+        {
+            for ( int i = 0; i < 100; i++ )
+            {
+                try ( Transaction transaction = database.beginTx() )
+                {
```

Comment:
Should inner loop be inside transaction?

Reply:
for some reason two for loops that the way of doing a lot of nodes, combined into one

Reply:
More is always better, right ;)